### ISSUE TITLE
PowerShell 2020.9.0 Integrated Console in a container: The language service could not be started

### ISSUE THREAD

**Issue by ebmarquez** [PowerShell/vscode-powershell#3058]
Docker Container

Issue Type: <b>Bug</b>

```
11/12/2020 6:42:56 PM [NORMAL] - Visual Studio Code v1.51.1 64-bit
11/12/2020 6:42:56 PM [NORMAL] - PowerShell Preview Extension v2020.9.0
11/12/2020 6:42:56 PM [NORMAL] - Operating System: Linux 64-bit
11/12/2020 6:42:56 PM [NORMAL] - Language server starting --
11/12/2020 6:42:56 PM [NORMAL] -     PowerShell executable: /usr/bin/pwsh
11/12/2020 6:42:56 PM [NORMAL] -     PowerShell args: -NoProfile -NonInteractive -EncodedCommand SQBtAHAAbwByAHQALQBNAG8AZAB1AGwAZQAgACcALwByAG8AbwB0AC8ALgB2AHMAYwBvAGQAZQAtAHMAZQByAHYAZQByAC8AZQB4AHQAZQBuAHMAaQBvAG4AcwAvAG0AcwAtAHYAcwBjAG8AZABlAC4AcABvAHcAZQByAHMAaABlAGwAbAAtAHAAcgBlAHYAaQBlAHcALQAyADAAMgAwAC4AOQAuADAALwBtAG8AZAB1AGwAZQBzAC8AUABvAHcAZQByAFMAaABlAGwAbABFAGQAaQB0AG8AcgBTAGUAcgB2AGkAYwBlAHMALwBQAG8AdwBlAHIAUwBoAGUAbABsAEUAZABpAHQAbwByAFMAZQByAHYAaQBjAGUAcwAuAHAAcwBkADEAJwA7ACAAUwB0AGEAcgB0AC0ARQBkAGkAdABvAHIAUwBlAHIAdgBpAGMAZQBzACAALQBIAG8AcwB0AE4AYQBtAGUAIAAnAFYAaQBzAHUAYQBsACAAUwB0AHUAZABpAG8AIABDAG8AZABlACAASABvAHMAdAAnACAALQBIAG8AcwB0AFAAcgBvAGYAaQBsAGUASQBkACAAJwBNAGkAYwByAG8AcwBvAGYAdAAuAFYAUwBDAG8AZABlACcAIAAtAEgAbwBzAHQAVgBlAHIAcwBpAG8AbgAgACcAMgAwADIAMAAuADkALgAwACcAIAAtAEEAZABkAGkAdABpAG8AbgBhAGwATQBvAGQAdQBsAGUAcwAgAEAAKAAnAFAAbwB3AGUAcgBTAGgAZQBsAGwARQBkAGkAdABvAHIAUwBlAHIAdgBpAGMAZQBzAC4AVgBTAEMAbwBkAGUAJwApACAALQBCAHUAbgBkAGwAZQBkAE0AbwBkAHUAbABlAHMAUABhAHQAaAAgACcALwByAG8AbwB0AC8ALgB2AHMAYwBvAGQAZQAtAHMAZQByAHYAZQByAC8AZQB4AHQAZQBuAHMAaQBvAG4AcwAvAG0AcwAtAHYAcwBjAG8AZABlAC4AcABvAHcAZQByAHMAaABlAGwAbAAtAHAAcgBlAHYAaQBlAHcALQAyADAAMgAwAC4AOQAuADAALwBtAG8AZAB1AGwAZQBzACcAIAAtAEUAbgBhAGIAbABlAEMAbwBuAHMAbwBsAGUAUgBlAHAAbAAgAC0AUwB0AGEAcgB0AHUAcABCAGEAbgBuAGUAcgAgACcAPQA9AD0APQA9AD4AIABQAG8AdwBlAHIAUwBoAGUAbABsACAAUAByAGUAdgBpAGUAdwAgAEkAbgB0AGUAZwByAGEAdABlAGQAIABDAG8AbgBzAG8AbABlACAAdgAyADAAMgAwAC4AOQAuADAAIAA8AD0APQA9AD0APQAKACcAIAAtAEwAbwBnAEwAZQB2AGUAbAAgACcATgBvAHIAbQBhAGwAJwAgAC0ATABvAGcAUABhAHQAaAAgACcALwByAG8AbwB0AC8ALgB2AHMAYwBvAGQAZQAtAHMAZQByAHYAZQByAC8AZQB4AHQAZQBuAHMAaQBvAG4AcwAvAG0AcwAtAHYAcwBjAG8AZABlAC4AcABvAHcAZQByAHMAaABlAGwAbAAtAHAAcgBlAHYAaQBlAHcALQAyADAAMgAwAC4AOQAuADAALwBsAG8AZwBzAC8AMQA2ADAANQAyADAANgA1ADcANgAtAGYAYQBmADMAZQAwADYAYQAtAGUAYgBjADcALQA0AGQAZAA2AC0AOQBhADIAZAAtADEAYwBhAGIAYwBmADEAMQA5ADIAMgA5ADEANgAwADUAMgAwADYANAA4ADAAOQA1ADUALwBFAGQAaQB0AG8AcgBTAGUAcgB2AGkAYwBlAHMALgBsAG8AZwAnACAALQBTAGUAcwBzAGkAbwBuAEQAZQB0AGEAaQBsAHMAUABhAHQAaAAgACcALwByAG8AbwB0AC8ALgB2AHMAYwBvAGQAZQAtAHMAZQByAHYAZQByAC8AZQB4AHQAZQBuAHMAaQBvAG4AcwAvAG0AcwAtAHYAcwBjAG8AZABlAC4AcABvAHcAZQByAHMAaABlAGwAbAAtAHAAcgBlAHYAaQBlAHcALQAyADAAMgAwAC4AOQAuADAALwBzAGUAcwBzAGkAbwBuAHMALwBQAFMARQBTAC0AVgBTAEMAbwBkAGUALQB1AG4AZABlAGYAaQBuAGUAZAAtADcAMQAzADEAMAA1ACcAIAAtAEYAZQBhAHQAdQByAGUARgBsAGEAZwBzACAAQAAoACkAIAA=
11/12/2020 6:42:56 PM [NORMAL] -     PowerShell Editor Services args: Import-Module '/root/.vscode-server/extensions/ms-vscode.powershell-preview-2020.9.0/modules/PowerShellEditorServices/PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2020.9.0' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath '/root/.vscode-server/extensions/ms-vscode.powershell-preview-2020.9.0/modules' -EnableConsoleRepl -StartupBanner '=====> PowerShell Preview Integrated Console v2020.9.0 <=====
' -LogLevel 'Normal' -LogPath '/root/.vscode-server/extensions/ms-vscode.powershell-preview-2020.9.0/logs/1605206576-faf3e06a-ebc7-4dd6-9a2d-1cabcf1192291605206480955/EditorServices.log' -SessionDetailsPath '/root/.vscode-server/extensions/ms-vscode.powershell-preview-2020.9.0/sessions/PSES-VSCode-undefined-713105' -FeatureFlags @() 
11/12/2020 6:42:56 PM [NORMAL] - pwsh started.
11/12/2020 6:42:56 PM [NORMAL] - Waiting for session file
11/12/2020 6:43:56 PM [NORMAL] - Timed out waiting for session file to appear.
11/12/2020 6:43:56 PM [NORMAL] - Language server startup failed.
11/12/2020 6:43:56 PM [ERROR] - The language service could not be started: 
11/12/2020 6:43:56 PM [ERROR] - Error: Timed out waiting for session file to appear.
11/12/2020 6:44:35 PM [NORMAL] - Shutting down language client...
11/12/2020 6:44:35 PM [NORMAL] - Language server starting --
11/12/2020 6:44:35 PM [NORMAL] -     PowerShell executable: /usr/bin/pwsh
11/12/2020 6:44:35 PM [NORMAL] -     PowerShell args: -NoProfile -NonInteractive -EncodedCommand SQBtAHAAbwByAHQALQBNAG8AZAB1AGwAZQAgACcALwByAG8AbwB0AC8ALgB2AHMAYwBvAGQAZQAtAHMAZQByAHYAZQByAC8AZQB4AHQAZQBuAHMAaQBvAG4AcwAvAG0AcwAtAHYAcwBjAG8AZABlAC4AcABvAHcAZQByAHMAaABlAGwAbAAtAHAAcgBlAHYAaQBlAHcALQAyADAAMgAwAC4AOQAuADAALwBtAG8AZAB1AGwAZQBzAC8AUABvAHcAZQByAFMAaABlAGwAbABFAGQAaQB0AG8AcgBTAGUAcgB2AGkAYwBlAHMALwBQAG8AdwBlAHIAUwBoAGUAbABsAEUAZABpAHQAbwByAFMAZQByAHYAaQBjAGUAcwAuAHAAcwBkADEAJwA7ACAAUwB0AGEAcgB0AC0ARQBkAGkAdABvAHIAUwBlAHIAdgBpAGMAZQBzACAALQBIAG8AcwB0AE4AYQBtAGUAIAAnAFYAaQBzAHUAYQBsACAAUwB0AHUAZABpAG8AIABDAG8AZABlACAASABvAHMAdAAnACAALQBIAG8AcwB0AFAAcgBvAGYAaQBsAGUASQBkACAAJwBNAGkAYwByAG8AcwBvAGYAdAAuAFYAUwBDAG8AZABlACcAIAAtAEgAbwBzAHQAVgBlAHIAcwBpAG8AbgAgACcAMgAwADIAMAAuADkALgAwACcAIAAtAEEAZABkAGkAdABpAG8AbgBhAGwATQBvAGQAdQBsAGUAcwAgAEAAKAAnAFAAbwB3AGUAcgBTAGgAZQBsAGwARQBkAGkAdABvAHIAUwBlAHIAdgBpAGMAZQBzAC4AVgBTAEMAbwBkAGUAJwApACAALQBCAHUAbgBkAGwAZQBkAE0AbwBkAHUAbABlAHMAUABhAHQAaAAgACcALwByAG8AbwB0AC8ALgB2AHMAYwBvAGQAZQAtAHMAZQByAHYAZQByAC8AZQB4AHQAZQBuAHMAaQBvAG4AcwAvAG0AcwAtAHYAcwBjAG8AZABlAC4AcABvAHcAZQByAHMAaABlAGwAbAAtAHAAcgBlAHYAaQBlAHcALQAyADAAMgAwAC4AOQAuADAALwBtAG8AZAB1AGwAZQBzACcAIAAtAEUAbgBhAGIAbABlAEMAbwBuAHMAbwBsAGUAUgBlAHAAbAAgAC0AUwB0AGEAcgB0AHUAcABCAGEAbgBuAGUAcgAgACcAPQA9AD0APQA9AD4AIABQAG8AdwBlAHIAUwBoAGUAbABsACAAUAByAGUAdgBpAGUAdwAgAEkAbgB0AGUAZwByAGEAdABlAGQAIABDAG8AbgBzAG8AbABlACAAdgAyADAAMgAwAC4AOQAuADAAIAA8AD0APQA9AD0APQAKACcAIAAtAEwAbwBnAEwAZQB2AGUAbAAgACcARABpAGEAZwBuAG8AcwB0AGkAYwAnACAALQBMAG8AZwBQAGEAdABoACAAJwAvAHIAbwBvAHQALwAuAHYAcwBjAG8AZABlAC0AcwBlAHIAdgBlAHIALwBlAHgAdABlAG4AcwBpAG8AbgBzAC8AbQBzAC0AdgBzAGMAbwBkAGUALgBwAG8AdwBlAHIAcwBoAGUAbABsAC0AcAByAGUAdgBpAGUAdwAtADIAMAAyADAALgA5AC4AMAAvAGwAbwBnAHMALwAxADYAMAA1ADIAMAA2ADYANwA1AC0AZgBhAGYAMwBlADAANgBhAC0AZQBiAGMANwAtADQAZABkADYALQA5AGEAMgBkAC0AMQBjAGEAYgBjAGYAMQAxADkAMgAyADkAMQA2ADAANQAyADAANgA0ADgAMAA5ADUANQAvAEUAZABpAHQAbwByAFMAZQByAHYAaQBjAGUAcwAuAGwAbwBnACcAIAAtAFMAZQBzAHMAaQBvAG4ARABlAHQAYQBpAGwAcwBQAGEAdABoACAAJwAvAHIAbwBvAHQALwAuAHYAcwBjAG8AZABlAC0AcwBlAHIAdgBlAHIALwBlAHgAdABlAG4AcwBpAG8AbgBzAC8AbQBzAC0AdgBzAGMAbwBkAGUALgBwAG8AdwBlAHIAcwBoAGUAbABsAC0AcAByAGUAdgBpAGUAdwAtADIAMAAyADAALgA5AC4AMAAvAHMAZQBzAHMAaQBvAG4AcwAvAFAAUwBFAFMALQBWAFMAQwBvAGQAZQAtAHUAbgBkAGUAZgBpAG4AZQBkAC0ANwA3ADAAMwAyADAAJwAgAC0ARgBlAGEAdAB1AHIAZQBGAGwAYQBnAHMAIABAACgAKQAgAA==
11/12/2020 6:44:35 PM [NORMAL] -     PowerShell Editor Services args: Import-Module '/root/.vscode-server/extensions/ms-vscode.powershell-preview-2020.9.0/modules/PowerShellEditorServices/PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2020.9.0' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath '/root/.vscode-server/extensions/ms-vscode.powershell-preview-2020.9.0/modules' -EnableConsoleRepl -StartupBanner '=====> PowerShell Preview Integrated Console v2020.9.0 <=====
' -LogLevel 'Diagnostic' -LogPath '/root/.vscode-server/extensions/ms-vscode.powershell-preview-2020.9.0/logs/1605206675-faf3e06a-ebc7-4dd6-9a2d-1cabcf1192291605206480955/EditorServices.log' -SessionDetailsPath '/root/.vscode-server/extensions/ms-vscode.powershell-preview-2020.9.0/sessions/PSES-VSCode-undefined-770320' -FeatureFlags @() 
11/12/2020 6:44:35 PM [NORMAL] - pwsh started.
11/12/2020 6:44:35 PM [NORMAL] - Waiting for session file
11/12/2020 6:45:35 PM [NORMAL] - Timed out waiting for session file to appear.
11/12/2020 6:45:35 PM [NORMAL] - Language server startup failed.
11/12/2020 6:45:35 PM [ERROR] - The language service could not be started: 
11/12/2020 6:45:35 PM [ERROR] - Error: Timed out waiting for session file to appear.
```

Extension version: 2020.9.0
VS Code version: Code 1.51.1 (e5a624b788d92b8d34d1392e4c4d9789406efe8f, 2020-11-10T23:34:32.027Z)
OS version: Windows_NT x64 10.0.19042
Remote OS version: Linux x64 4.19.128-microsoft-standard

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz (8 x 3408)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|63.92GB (44.64GB free)|
|Process Argv|--crash-reporter-id 4e4350ed-2c2e-4a28-b179-34363158d8c6|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Container partnertoolkit:latest (/keen_…|
|OS|Linux x64 4.19.128-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz (8 x 3407)|
|Memory (System)|50.15GB (46.74GB free)|
|VM|0%|
</details>

**Comment by ebmarquez** [comment]
Reverted back to v2020.6.0 and it worked as expected.

**Comment by SydneyhSmith** [comment]
Thanks @ebmarquez would you mind sharing your docker file with us?

**Comment by ebmarquez** [comment]
Here are the docker details.  The only error the IDE provided was a timeout starting the PowerShell Integrated Console.

1. start the container in interactive mode to keep it running.
2. Using Remote-Container I attach VSCode to the running container.
3. Open the Workspace folder /app in the IDE.
4. Install the PowerShell Preview v2020.9.0
5. Select Default shell as /usr/bin/pwsh
6. Close the VSCode that is attached to the container
7. Attach a new VSCode to the still running container.  This will startup with the PowerShell Integrated Console by default.  This is where it fails.



```
# Based on DOTNET image, which supports ASPNET CORE runtime
FROM mcr.microsoft.com/dotnet/core/aspnet:3.1 AS base

WORKDIR /app
ENV DEBIAN_VERSION=buster

# Disable some unnecessary warning message during the image build
ARG DEBIAN_FRONTEND=noninteractive

# Install PowerShell module in the image
RUN apt-get update \
    && apt-get install -y gnupg \
    && apt-get install -y --no-install-recommends \
    apt-utils \
    ca-certificates \
    curl \
    apt-transport-https \
    && curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - \
    && echo "deb [arch=amd64] https://packages.microsoft.com/repos/microsoft-debian-${DEBIAN_VERSION}-prod ${DEBIAN_VERSION} main" > /etc/apt/sources.list.d/microsoft.list \
    && apt-get update \
    && apt-get install -y --no-install-recommends \
    software-properties-common \
    powershell=7.0.0-1.debian.10

# Set up python chroot - this saves considerable space over a system-wide installation
# by using python's native dependency management and only installing minimal system packages
RUN apt-get install -y --no-install-recommends \
    python3-pkg-resources \
    python3-virtualenv \
    virtualenv

# Set up python chroot then install ansible inside it
ENV VIRTUAL_ENV=/app/venv
RUN python3 -m virtualenv --python=/usr/bin/python3 $VIRTUAL_ENV
ENV PATH="$VIRTUAL_ENV/bin:$PATH"
RUN pip install \
    ansible==2.10.0 \
    jinja2==2.11.2 \
    cffi==1.14.0 \
    cryptography==2.9.2 \
    MarkupSafe==1.1.1 \
    netaddr==0.7.19 \
    pkg-resources==0.0.0 \
    pycparser==2.20 \
    PyYAML==5.3.1 \
    six==1.14.0 \
    paramiko==2.7.1

# Clean up apt caches
RUN apt-get autoremove \
    && rm -rf /var/lib/apt/lists/*

# Install Powershell Tools
RUN pwsh -c 'Install-Module powershell-yaml -MaximumVersion 0.4.2 -Scope AllUsers -Force -ErrorAction Stop'
```

## Dev Containers log

```log
[10 ms] Remote-Containers 0.148.1 in VS Code 1.51.1 (e5a624b788d92b8d34d1392e4c4d9789406efe8f).
[24 ms] Start: Resolving remote
[25 ms] Setting up container /wizardly_rosalind

[26 ms] Start: Run: docker inspect --type container /wizardly_rosalind
[626 ms] Start: Check Docker is running
[627 ms] Start: Run: docker info
[3114 ms] Start: Inspecting container
[3114 ms] Start: Run: docker inspect --type container 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0
[3625 ms] Start: Run: docker exec -i -u root -e VSCODE_REMOTE_CONTAINERS_SESSION=65b87b9a-84c6-4dc1-ac96-e1a70a8df5c71605223844894 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh
[3643 ms] Start: Run in container: uname -m
[4203 ms] x86_64
[4204 ms] 
[4204 ms] Start: Run in container: (cat /etc/os-release || cat /usr/lib/os-release) 2>/dev/null
[4206 ms] PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
[4207 ms] 
[4207 ms] Start: Run in container: cat /etc/passwd
[4210 ms] Start: Run in container: test -d /root/.vscode-server
[4212 ms] 
[4213 ms] 
[4213 ms] Start: Run in container: set -o noclobber ; mkdir -p '/root/.vscode-server/data/Machine' && { > '/root/.vscode-server/data/Machine/.writeMachineSettingsMarker' ; } 2> /dev/null
[4216 ms] 
[4216 ms] 
[4216 ms] Exit code 2
[4217 ms] Start: Run in container: test -d /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f
[4219 ms] 
[4219 ms] 
[4219 ms] Start: Launching Remote-Containers helper.
[4220 ms] Start: Run: gpgconf --list-dir agent-extra-socket
[4240 ms] Start: Run in container: cat <<'EOF-/tmp/vscode-remote-containers-02b2204872f7860e7c5e7ce099f9317db789e40f.js' >/tmp/vscode-remote-containers-02b2204872f7860e7c5e7ce099f9317db789e40f.js
[4246 ms] 
[4247 ms] 
[4247 ms] Start: Run in container: # Test for /root/.ssh/known_hosts and ssh
[4249 ms] ssh not found
[4249 ms] 
[4249 ms] Exit code 1
[4249 ms] Start: Run in container: cat <<'EOF-/tmp/vscode-remote-containers-server-02b2204872f7860e7c5e7ce099f9317db789e40f.js' >/tmp/vscode-remote-containers-server-02b2204872f7860e7c5e7ce099f9317db789e40f.js
[4253 ms] 
[4253 ms] 
[4253 ms] Start: Run in container: command -v git >/dev/null 2>&1 && git config --system credential.helper '!f() { /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f/node /tmp/vscode-remote-containers-02b2204872f7860e7c5e7ce099f9317db789e40f.js $*; }; f' || true
[4253 ms] Start: Run: docker exec -i -u root -e REMOTE_CONTAINERS_SOCKETS=["/tmp/vscode-ssh-auth-02b2204872f7860e7c5e7ce099f9317db789e40f.sock"] -e REMOTE_CONTAINERS_IPC=/tmp/vscode-remote-containers-ipc-02b2204872f7860e7c5e7ce099f9317db789e40f.sock 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f/node /tmp/vscode-remote-containers-server-02b2204872f7860e7c5e7ce099f9317db789e40f.js
[4255 ms] 
[4255 ms] 
[4255 ms] Start: Run in container: set -o noclobber ; mkdir -p '/root/.vscode-server/data/Machine' && { > '/root/.vscode-server/data/Machine/.installExtensionsMarker' ; } 2> /dev/null
[4259 ms] 
[4259 ms] 
[4259 ms] Exit code 2
[4261 ms] Start: Run in container: # Test for /root/.gitconfig and git
[4263 ms] git not found
[4263 ms] 
[4263 ms] Exit code 1
[4263 ms] Start: Run in container: command -v git >/dev/null 2>&1 && git config --global credential.helper '!f() { /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f/node /tmp/vscode-remote-containers-02b2204872f7860e7c5e7ce099f9317db789e40f.js $*; }; f' || true
[4266 ms] 
[4266 ms] 
[4266 ms] Start: Run in container: for pid in `cd /proc && ls -d [0-9]*`; do { echo $pid ; readlink -f /proc/$pid/cwd ; xargs -0 < /proc/$pid/environ ; xargs -0 < /proc/$pid/cmdline ; } ; echo ; done 2>/dev/null
[4318 ms] Extension host agent is already running.
[4318 ms] Start: Run in container: cat /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f/.devport 2>/dev/null
[4321 ms] 44589
[4321 ms] 
[4325 ms] Forwarding local port 44589 to container port 44589
[4337 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/extensions 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch installed extensions
[4338 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/data/Machine 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch machine settings
[4353 ms] Start: Run: docker exec -i -u root -e VSCODE_REMOTE_CONTAINERS_SESSION=65b87b9a-84c6-4dc1-ac96-e1a70a8df5c71605223844894 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f/node -e 
[4354 ms] Start: Run: docker exec -i -u root -e VSCODE_REMOTE_CONTAINERS_SESSION=65b87b9a-84c6-4dc1-ac96-e1a70a8df5c71605223844894 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f/node -e 
[28457 ms] Start: Run: docker exec -i -u root -e VSCODE_AGENT_FOLDER=/root/.vscode-server -w /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f/server.sh --list-extensions
[29408 ms] 
[29408 ms] 
[29409 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/extensions 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch installed extensions
[31122 ms] Start: Run: docker exec -i -u root -e VSCODE_AGENT_FOLDER=/root/.vscode-server -w /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f/server.sh --list-extensions
[32061 ms] 
[32061 ms] 
[32061 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/extensions 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch installed extensions
[32717 ms] Start: Run: docker exec -i -u root -e VSCODE_REMOTE_CONTAINERS_SESSION=65b87b9a-84c6-4dc1-ac96-e1a70a8df5c71605223844894 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f/node -e 
[33693 ms] Start: Run: docker exec -i -u root -e VSCODE_AGENT_FOLDER=/root/.vscode-server -w /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /root/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f/server.sh --list-extensions
[34473 ms] ms-vscode.powershell-preview
[34473 ms] 
[34477 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/extensions 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch installed extensions
[66474 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/data/Machine 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch machine settings
[95398 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/extensions 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch installed extensions
[128326 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/data/Machine 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch machine settings
[158206 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/extensions 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch installed extensions
[189147 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/data/Machine 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch machine settings
[219103 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/extensions 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch installed extensions
[250042 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/data/Machine 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch machine settings
[280029 ms] Start: Run: docker exec -i -u root -w /root/.vscode-server/extensions 9f7c7197c5149326652dcf996f70c61b35ae2cdcf0997a63be25c59a1a18fdb0 /bin/sh -c # Watch installed extensions
```

**Comment by SydneyhSmith** [comment]
Thanks @ebmarquez this is really helpful information! Are you able to start PowerShell as a non-integrated console in VSCode? 
![image](https://user-images.githubusercontent.com/43417619/99731007-1876a880-2a72-11eb-97bd-21b656d82850.png)
It would be really helpful if you could also attach the diagnostic logs (rather than normal logging)--instructions for how to do that are [here ](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#providing-information-about-your-environment), or any other log files you may have

**Comment by ebmarquez** [comment]
I can open a new non-integrated console and use pwsh.  This was not as helpful as I would like since I wanted to debug my script that operates in the container. I had a few breakpoints I needed to hit.  breakpoints are nice to have! :-)

I zipped up the PowerShell folder that ran in the container.
[ms-vscode.powershell-preview-2020.9.0.tar.gz](https://github.com/PowerShell/vscode-powershell/files/5570121/ms-vscode.powershell-preview-2020.9.0.tar.gz)
